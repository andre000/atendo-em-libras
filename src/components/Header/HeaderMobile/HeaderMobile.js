import React, { PureComponent } from 'react'
import styled from 'styled-components'
import { PrimaryButton } from '../../Buttons'
import { Layer } from 'grommet'

const StyledButton = styled(PrimaryButton)`
  border-radius: 20px 20px 20px 0;
`

const MainMenuModal = styled(Layer)`
  background-color: red;
`

class HeaderMobile extends PureComponent {
  render = () => {
    return (
      <MainMenuModal modal plain full>
        <a target="_blank" rel="noopener noreferrer" href="https://forms.gle/h1sX9nD45PgUAzGB6">
          <StyledButton>Cadastre Aqui</StyledButton>
        </a>
      </MainMenuModal>
    )
  }
}

export { HeaderMobile }
