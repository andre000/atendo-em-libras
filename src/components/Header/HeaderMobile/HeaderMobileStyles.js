import styled from 'styled-components'
import { PrimaryButton } from '../../Buttons'
import { Layer } from 'grommet'

const StyledButton = styled(PrimaryButton)`
  border-radius: 20px 20px 20px 0;
  background-color: white;
  color: #5996f7;
`

// height and position are handled manually to prevent
// fullscreen modal issue on mobile browsers. For more info read:
// https://css-tricks.com/the-trick-to-viewport-units-on-mobile/
const MainMenuModal = styled(Layer)`
  position: fixed;
  height: 100%;
  background-color: #5996f7;
`

const Logo = styled.img`
  height: auto;
  width: 100px;
`

const Wrapper = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: space-between;
  padding: 30px;
`

const Header = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`

const Footer = styled.div`
  h2 {
    width: 80%;
    color: white;
    font-size: 1.8rem;
    font-weight: 400;
    line-height: 1.6em;
    margin-bottom: 21px;
  }
`

export { StyledButton, MainMenuModal, Logo, Wrapper, Header, Footer }
