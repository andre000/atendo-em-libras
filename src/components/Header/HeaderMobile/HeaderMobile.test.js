import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { HeaderMobile } from './HeaderMobile'

describe('HeaderMobile component', () => {
  let view
  let onClose

  beforeEach(() => {
    onClose = jest.fn()
    view = render(<HeaderMobile onClose={onClose} />)
  })

  it('should execute callback function when close button is clicked', () => {
    const closeButton = view.getByLabelText('Close Menu Button')

    expect(closeButton).toBeInTheDocument()
    fireEvent.click(closeButton)
    expect(onClose).toHaveBeenCalled()
  })

  it('should redirect to form when register button is clicked', () => {
    const registerLink = view.getByText('Cadastre-se aqui')

    expect(registerLink.href).toBe('https://forms.gle/h1sX9nD45PgUAzGB6')
  })

  it('Render correctly', () => {
    const { baseElement } = view
    expect(baseElement).toMatchSnapshot()
  })
})
