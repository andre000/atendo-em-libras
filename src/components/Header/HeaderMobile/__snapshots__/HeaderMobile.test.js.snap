// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMobile component Render correctly 1`] = `
.c8 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c11 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c11:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: transparent;
  position: relative;
  z-index: 10;
  pointer-events: none;
  outline: none;
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
}

.c1 {
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  pointer-events: all;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 48px;
  outline: none;
  pointer-events: all;
  z-index: 15;
  position: absolute;
  max-height: calc(100% - 0px - 0px);
  max-width: calc(100% - 0px - 0px);
  border-radius: 0;
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
  -webkit-animation: gVgDGd 0.2s ease-in-out forwards;
  animation: gVgDGd 0.2s ease-in-out forwards;
}

.c4 {
  width: 0;
  height: 0;
  overflow: hidden;
  position: absolute;
}

.c9 {
  font-size: 1.4rem;
  font-weight: 600;
}

.c12 {
  background-color: #5996f7;
  border-radius: 17px;
  color: white;
  border-radius: 20px 20px 20px 0;
  background-color: white;
  color: #5996f7;
}

.c12:hover {
  background-color: #96bfff;
}

.c12:focus {
  outline: none;
  outline-offset: 0;
}

.c3 {
  position: fixed;
  height: 100%;
  background-color: #5996f7;
}

.c7 {
  height: auto;
  width: 100px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 30px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 h2 {
  width: 80%;
  color: white;
  font-size: 1.8rem;
  font-weight: 400;
  line-height: 1.6em;
  margin-bottom: 21px;
}

@media only screen and (max-width:768px) {
  .c0 {
    position: absolute;
    width: 100%;
    height: 100%;
    min-height: 100vh;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    position: relative;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    position: relative;
    max-height: none;
    max-width: none;
    border-radius: 0;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    -webkit-transform: none;
    -ms-transform: none;
    transform: none;
    -webkit-animation: none;
    animation: none;
    height: 100vh;
    width: 100vw;
  }
}

<body>
  <div
    aria-hidden="false"
  />
  <div
    aria-hidden="false"
    id="layerClone"
  >
    <div
      aria-hidden="false"
    >
      <div
        class="c0"
        tabindex="-1"
      >
        <div
          class="c1"
        />
      </div>
    </div>
  </div>
  <div />
  <div
    aria-hidden="false"
    id="layerClone"
  >
    <div
      aria-hidden="false"
    >
      <div
        class="c0"
        tabindex="-1"
      >
        <div
          class="c1"
        />
      </div>
    </div>
  </div>
  <div />
  <div>
    <div
      aria-hidden="false"
    >
      <div
        class="c0"
        tabindex="-1"
      >
        <div
          class="c1"
        />
        <div
          class="c2 c3"
        >
          <a
            aria-hidden="true"
            class="c4"
            tabindex="-1"
          />
          <div
            class="c5"
            role="menu"
          >
            <div
              class="c6"
            >
              <img
                alt="Logo projeto Atendo em Libras"
                class="c7"
                src="img_atendo_em_libras_logo_all_white.svg"
              />
              <button
                aria-label="Close Menu Button"
                class="c8 c9"
                type="button"
              >
                <img
                  alt="Close Icon"
                  height="18px"
                  src="ic_close.svg"
                  width="18px"
                />
              </button>
            </div>
            <div
              class="c10"
            >
              <h2>
                É profissional de saúde ou conhece alguém da área?
              </h2>
              <a
                aria-label="register form"
                class="c11 c9 Buttons__PrimaryButton-nr8h4a-1 c12"
                href="https://forms.gle/h1sX9nD45PgUAzGB6"
                rel="noopener noreferrer"
                target="_blank"
              >
                Cadastre-se aqui
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
